addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusInverterRealtimeDataHandler.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fronius.handler;
+
+import static org.openhab.binding.fronius.FroniusBindingConstants.*;
+
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.fronius.internal.model.InverterRealtimeData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Handler for Fronius Inverter Realtime Data.
+ *
+ * @author Gerrit Beine - Initial contribution
+ */
+public class FroniusInverterRealtimeDataHandler extends FroniusDeviceThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FroniusInverterRealtimeDataHandler.class);
+    private final JsonParser parser = new JsonParser();
+
+    private String url;
+
+    public FroniusInverterRealtimeDataHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected String getServiceDescription() {
+        return INVERTER_REALTIME_DATA_DESCRIPTION;
+    }
+
+    @Override
+    protected String getServiceUrl() {
+        if (null == url) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("http://");
+            sb.append(getHostname());
+            sb.append(INVERTER_REALTIME_DATA_URL);
+            sb.append("?Scope=Device");
+            sb.append("&DeviceId=");
+            sb.append(getDevice());
+            sb.append("&DataCollection=CommonInverterData");
+            url = sb.toString();
+        }
+
+        logger.debug("{} URL: {}", getServiceDescription(), url);
+
+        return url;
+    }
+
+    @Override
+    protected void updateData(String data) {
+        logger.debug("Refresh data {}", data);
+        final JsonObject json = parser.parse(data).getAsJsonObject();
+        final InverterRealtimeData model = InverterRealtimeData.createInverterRealtimeData(json);
+        if (model.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        } else {
+            updateState(new ChannelUID(getThing().getUID(), CHANNEL_INVERTER_DAY_ENERGY), model.getDayEnergy());
```

Comment:
Instead of creating new ChannelUID objects each time you probably better loop over the channels via `for (Channel channel : getThing().getChannels()) {` and in a switch statement handle each specific state.